src/state/learnocaml_store.ml

```diff
@@ -203,7 +203,12 @@ module Exercise = struct
           List.iter (fun st -> Hashtbl.add tbl st.id st) l;
           tbl)
       @@ function
-      | Unix.Unix_error (Unix.ENOENT, _, _) -> Lwt.return tbl
+      | Unix.Unix_error (Unix.ENOENT, _, _) ->
+          Lazy.force !index >>= fun index ->
+          Exercise.Index.fold_exercises (fun () id _ ->
+              Hashtbl.add tbl id (Exercise.Status.default id))
+            () index;
+          Lwt.return tbl
```

Comment:
Hi, thanks a lot @AltGr for your PR!

Just to be sure, do you confirm that this code fixing #314|#337 won't have a side-effect of allowing students that would be already connected to a Learn-OCaml instance URL, to effectively connect to a new server at the same URL if the teacher has reset the `/sync` folder? (because in this case, it would be a small flaw regarding authorization - assuming the instance requires a new "secret" phrase for creating tokens)

Anyway, I think this issue does not occur as the two features are independent, but just wanted to double-check with you 😅

Reply:
This only concerns the exercise status index (as opposed to the student index) so I don't believe it could have any consequence on the tokens. Previously, when the exercise status index didn't exist, exercises would be assumed open, but the index itself was assumed empty, which led to the status of students not being displayed correctly on the teacher screen.